source/base/index_set.cc

```diff
@@ -547,9 +546,66 @@ IndexSet::make_trilinos_map (const MPI_Comm &communicator,
                         );
     }
 }
+#endif
 
 
-#endif
+
+bool
+IndexSet::is_globally_ascending (const MPI_Comm &communicator) const
+{
+  // If there is only one process, the IndexSet is always ascending.
+  const unsigned int n_ranks = Utilities::MPI::n_mpi_processes(communicator);
+  if (n_ranks==1)
+    return true;
+  // Non-contiguous IndexSets can't be linear.
+  const bool all_contiguous = (Utilities::MPI::min (is_contiguous() ? 1 : 0, communicator) == 1);
+  if (!all_contiguous)
+    return false;
+
+  bool is_globally_ascending = true;
+  // we know that there is only one interval
+  types::global_dof_index local_dofs[2];
+  local_dofs[0] = (n_elements()>0) ? *(begin_intervals()->begin())
+                  : numbers::invalid_dof_index ;
+  local_dofs[1] = (n_elements()>0) ? begin_intervals()->last()
+                  : numbers::invalid_dof_index;
+
+  const unsigned int my_rank = Utilities::MPI::this_mpi_process(communicator);
+  // first gather all information on process 0
+  const unsigned int gather_size = (my_rank==0)?2*n_ranks:1;
+  std::vector<types::global_dof_index> global_dofs(gather_size);
+
+  MPI_Gather(local_dofs, 2, DEAL_II_DOF_INDEX_MPI_TYPE,
+             &(global_dofs[0]), 2, DEAL_II_DOF_INDEX_MPI_TYPE, 0,
+             communicator);
+  if (my_rank == 0)
+    {
+      // find out if the received std::vector is ascending
+      types::global_dof_index old_dof = global_dofs[0], new_dof = 0;
+      types::global_dof_index index = 0;
+      while (global_dofs[index] == numbers::invalid_dof_index)
+        ++index;
+      old_dof = global_dofs[index++];
+      for (; index<global_dofs.size(); ++index)
+        {
+          new_dof = global_dofs[index];
+          if (new_dof == numbers::invalid_dof_index)
+            new_dof = old_dof;
+          else if (new_dof<old_dof)
+            {
+              is_globally_ascending = false;
+              break;
+            }
+        }
+    }
+
+  // now broadcast the result
+  int is_ascending = is_globally_ascending ? 1 : 0;
+  MPI_Bcast(&is_ascending, 1, MPI_INT, 0, communicator);
```

Comment:
In this case, I'd rather say that the chosen variant is faster because the `Gather` collects a vector as long as the number of processors, and so would `Allgather`, whereas the `Bcast` only sends around a scalar. Given the amount of data, avoiding the `Scatter` could matter for many processors. (Note it this is purely a throughput issue and I expect `Scatter` and `Bcast` to perform similarly with respect to latency.)